tools/clang/test/CodeGenSPIRV/cs.groupshared.not-in-globals.hlsl

```diff
@@ -0,0 +1,45 @@
+// Run: %dxc -T cs_6_0 -E main
+
+// A and B are groupshared, and should not be placed in the $Globals cbuffer.
+// myGlobalInteger is a global variable and is placed in $Globals cbuffer.
+
+// CHECK:    OpMemberName %type__Globals 0 "myGlobalInteger"
+// CHECK-NOT:OpMemberName %type__Globals 1
+
+// CHECK: OpMemberName %type_myCbuffer 0 "myA"
+// CHECK: OpMemberName %type_myCbuffer 1 "myB"
+// CHECK: OpMemberName %type_myCbuffer 2 "myC"
+
+// CHECK:      OpMemberDecorate %type__Globals 0 Offset
+// CHECK-NOT:  OpMemberDecorate %type__Globals 1 Offset
+// CHECK-NOT:  OpMemberDecorate %type__Globals 2 Offset
+
+// CHECK:      OpMemberDecorate %type_myCbuffer 0 Offset
+// CHECK-NEXT: OpMemberDecorate %type_myCbuffer 1 Offset
+// CHECK-NEXT: OpMemberDecorate %type_myCbuffer 2 Offset
+
+// CHECK: %type__Globals = OpTypeStruct %int
+// CHECK: %type_myCbuffer = OpTypeStruct %int %int %int
+
+// CHECK: %A = OpVariable %_ptr_Workgroup_int Workgroup
+groupshared int A;
+
+// CHECK: %B = OpVariable %_ptr_Workgroup__arr_int_uint_10 Workgroup
+groupshared int B[10];
+
+// CHECK: %_Globals = OpVariable %_ptr_Uniform_type__Globals Uniform
+int myGlobalInteger;
+
+// CHECK: %myCbuffer = OpVariable %_ptr_Uniform_type_myCbuffer Uniform
+cbuffer myCbuffer {
+  int myA;
+  int myB;
+  groupshared int myC;
```

Comment:
I wonder what is the correct definition of `myC` in SPIR-V code.
Maybe `// CHECK: %myC = OpVariable %_ptr_Workgroup_int Workgroup`?

It is weird to me that we specify `OpMemberName %type_myCbuffer 2 "myC"`, but we skipped the register assignment for `myC`.
Could you please add an example that actually affects the register assignment because of `groupshared`?